File: java/class 14/Magic.java
import java.util.Scanner;

public class Magic{
  public static void main(String[] args){
    Deck d = new Deck();
    for (int i = 0; i < 7; i++){
      d.shuffle();
    }
    Scanner in = new Scanner(System.in);
    int card_num = in.nextInt();
    System.out.println(d.cards.get(card_num).getName());
    Card c = d.cards.get(card_num);
    c.setMarked(true);
    for (int i = 0; i < 7; i++){
      d.shuffle();
    }
    System.out.println("abradacadabra");
    for (int i = 0; i < d.cards.size(); i++){
      if (d.cards.get(i).isMarked()){
        System.out.println(d.cards.get(i).getName());
        c.setMarked(false);
      }
    }
    in.close();
  } 
}

File: python/class 17-18/magic.py
from Deck import Deck
from Card import Card

d=Deck()
deck=d.cards

for i in range(7):
  d.shuffle()

card=int(input("Which card would you like to pick from a deck of 52 cards\n(state the card number example: 5 for the 5th card): "))
print(f"Your card is '{deck[card-1].getName()}'")

deck[card-1].setMarked()
for i in range(7):
  d.shuffle()

print("abracadabra")
for i in deck:
  if i.isMarked():
    i.setMarked(marked=False)
    print(f"'{i.getName()}' is your card")
  

